### PR TITLE
disable updateNameFromFirstLine from untitledTextEditorModel

### PR DESCRIPTION
**Pull request by smartguest**
untitledTextEditorModel is created without a file path before the Query Editor is instantiated, and it detects whenever the editor text changes. As a result, it will replace the header name with the first line of text in the editor. There is currently no way to alter this behavior outside of vscode.

This PR handles #9058 until updates to vscode have been made.

### CODE REVIEW

File: src/vs/workbench/common/editor/untitledTextEditorModel.ts [hunk 1]
```diff
@@ -242,7 +242,8 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 	}
 
 	private updateNameFromFirstLine(): void {
-		if (this.hasAssociatedFilePath) {
+		// {{SQL CARBON EDIT}} - For Query Editor, must not change name even without file path.
+		if (!this.hasAssociatedFilePath && this.hasAssociatedFilePath) {
```
Comment (Charles-Gagnon) [inline src/vs/workbench/common/editor/untitledTextEditorModel.ts]: I'm not understanding this change - don't we want to return here? The statement you have is always going to evaluate to false so it'll always use the first line text...

Comment (smartguest) [reply src/vs/workbench/common/editor/untitledTextEditorModel.ts]: woops, I meant either.

**Comment by Charles-Gagnon** [comment]
Also - this is still resulting in not displaying the description (connected status for query editors) :

Current Stable :

image

With your fix :

image

**Comment by smartguest** [comment]
So far the title labels still have disconnected labeled on them, just that the sizing of the tabs are completely shrunk so that it won't display properly

**Comment by kburtram** [comment]
@smartguest why do you think it's a tab sizing issue ?  when I look in the DOM inspector I don't see the connection state in the tab label string?  are you seeing it somewhere else?

Screen Shot 2020-02-06 at 12 04 03 PM

**Comment by smartguest** [comment]
I see that when the label is being created, disconnected is included with the title and label, but it seems to be removed at the end for some reason

**Comment by kburtram** [comment]
@smartguest thanks for tracking this down!  I'll build your branch and help test.

**Comment by kburtram** [comment]
@smartguest let's wait to merge this until after we discuss issues in the 1:30 meeting.

**Comment by kburtram** [comment]
@smartguest this seems to be working for me during local testing.  Thanks for fixing!

File: src/vs/workbench/test/common/editor/untitledTextEditor.test.ts [hunk 2]
```diff
@@ -303,24 +303,24 @@ suite('Workbench untitled text editors', () => {
 		model.dispose();
 	});
 
-	test('service#onDidChangeLabel', async () => {
-		const service = accessor.untitledTextEditorService;
-		const input = service.create();
+	// test('service#onDidChangeLabel', async () => { // {{SQL CARBON EDIT}} - Disable as untitledTextEditorModel test is disabled.
```
Comment (Charles-Gagnon) [inline src/vs/workbench/test/common/editor/untitledTextEditor.test.ts]: I'd prefer just doing test.skip instead - reduces the number of changes were making to vs code stuff

File: src/vs/workbench/test/common/editor/untitledTextEditor.test.ts [hunk 3]
```diff
@@ -383,70 +383,70 @@ suite('Workbench untitled text editors', () => {
 		assert.ok(counter > 1);
 	});
 
-	test('model#onDidChangeName and input name', async function () {
```
Comment (kburtram) [inline src/vs/workbench/test/common/editor/untitledTextEditor.test.ts]: use test skip flag instead of commenting out tests.

**Pull request merged**